basics.js

```diff
@@ -31,6 +34,15 @@ def greet(name, excited)
 end
 */
 
+var greet = function(name, excited){
+  var greeting = " Hi! " + name + " here! ";
+  if (excited === true){
+    return greeting(str.toUpperCase()); // not sure about this syntax. :(
```

Comment:
So what's not working here is that you're treating `greeting` as a function, and then passing in `str` as an argument which you're calling .toUpperCase() on. I see what you're thinking, but the correct syntax would be `greeting.toUpperCase()`